### PR TITLE
refactor: use latest cdc client

### PR DESCRIPTION
**Pull request by Emrehzl94**
- Upgraded CDC client to the latest version.
- Implemented required changes on the connector side to support the new CDC client.
- Sent transaction config to CDC client to set telemetry data.

### CODE REVIEW

**Comment by Emrehzl94** [comment]
[Add transaction config supplier to cdc client implementation](https://trello.com/c/tHfQRNxw/485-add-transaction-config-supplier-to-cdc-client-implementation)

File: source/src/main/kotlin/org/neo4j/connectors/kafka/source/Neo4jCdcTask.kt [hunk 1]
```diff
@@ -67,6 +67,7 @@ class Neo4jCdcTask : SourceTask() {
         CDCClient(
             config.driver,
             { sessionConfig },
+            { config.txConfig() },
```
Comment (ali-ince) [inline source/src/main/kotlin/org/neo4j/connectors/kafka/source/Neo4jCdcTask.kt:70]: if the returned object doesn't change over invocations, we can pre-construct it and return that object to reduce object instantiations, like what we did for `sessionConfig`.

File: source/src/main/kotlin/org/neo4j/connectors/kafka/source/SourceConfiguration.kt [hunk 2]
```diff
@@ -253,28 +254,34 @@ class SourceConfiguration(originals: Map<*, *>) :
...
-      pattern.withMetadata(metadata)
+    if (metadataKey.startsWith("$METADATA_KEY_TX_METADATA.")) {
+      val txMetadataKey = metadataKey.removePrefix("$METADATA_KEY_TX_METADATA.")
+
+      var txMetadata = patternTxMetadataMap[pattern]
+      if (txMetadata == null) {
+        txMetadata = mutableMapOf(txMetadataKey to value)
+      } else {
+        txMetadata[txMetadataKey] = value
+      }
```
Comment (ali-ince) [inline source/src/main/kotlin/org/neo4j/connectors/kafka/source/SourceConfiguration.kt:272]: could `getOrPut` or similar methods on map work here?

**Review by ali-ince — APPROVED** [review]

File: source/src/main/kotlin/org/neo4j/connectors/kafka/source/SourceConfiguration.kt (hunk 2 above)
Comment (Emrehzl94) [reply source/src/main/kotlin/org/neo4j/connectors/kafka/source/SourceConfiguration.kt:272]: `getOrPut` works very well for this case 👍

**Comment by venikkin** [comment]
[Report import job progress to the operator](https://trello.com/c/16iGKRHE/608-report-import-job-progress-to-the-operator)

**Pull request merged**